src/main/java/teammates/common/util/Sanitizer.java

```diff
@@ -314,24 +294,25 @@ private static String trimIfNotNull(String string) {
      * @param text
      * @return safer version of the text for XPath
      */
-    public static String convertStringForXPath(String text) {
+    public static String sanitizeStringForXPath(String text) {
         StringBuilder result = new StringBuilder();
-        int startPos = 0;
-        int i = 0;
-        while (i < text.length()) {
-            while (i < text.length() && text.charAt(i) != '\'') {
-                i++;
-            }
-            if (startPos < i) {
-                result.append('\'').append(text.substring(startPos, i)).append("',");
-                startPos = i;
+        int startOfChain = 0;
+        int textLength = text.length();
+        boolean isSingleQuotationChain = false;
+        // currentPos iterates through 0 to textLength (inclusive) to include last chain
```

Comment:
`currentPos iterates from 0 to textLength (inclusive)...`
Consider changing the phrasing; this comment is a little hard to understand